mathesar/utils/datafiles.py

```diff
@@ -114,18 +114,17 @@ def create_datafile(data):
         )
     else:
         max_level = data.get('max_level', 0)
-        sheet_name = data.get('sheet_name', '1')
+        sheet_index = data.get('sheet_index', 1)
```

Comment:
Why have default be 1 if the index is 0-indexed? This also conflicts with the analogous field in the `DataFile` django model. Is this an oversight?

Reply:
I was reducing the `sheet_index` value received from the API just a line below, so I thought that if the POST request doesn't have this value, we can assign it default value 1 and then reduce it.

On second thought, I agree that the above method is confusing. I have two solutions:
1. Keep the value 0-indexed everywhere and inform users to give sheet index based on 0-indexing.
2. Keep the value 1-indexed everywhere -- when asking users and when assigning default value (which would be 1). We'll just reduce it when we call `pandas.read_excel()`.

@dmos62 what are your thoughts?

Reply:
I think 0-indexing is standard and we should go with that.

Reply:
@dmos62 I have updated it now. PTAL. Thanks!